### ISSUE TITLE
Bug of chainer.Link.copy

### ISSUE THREAD

**Issue by kamo-naoyuki** [pfnet/chainer#1952]
Reproduce code:

```
import chainer.links as L
from chainer import Variable
import numpy as np

v = Variable(np.ones((128, 100), dtype=np.float32))
l1 = L.Linear(in_size=None, out_size=100)
l2 = l1.copy()
l1(v)
l2(v)

```
Error message:
AttributeError: 'Linear' object has no attribute 'W'

I think this is a bug of chainer.Link.copy method.

My Solution: Append copying of _uninitialized_params

```
   def copy(self):
        ret = copy.copy(self)
        ret._params = list(self._params)
        ret._persistent = list(self._persistent)
        ret._uninitialized_params = copy.deepcopy(self._uninitialized_params)
        ret.name = None
        d = ret.__dict__
        for name in ret._params:
            d[name] = copy.copy(d[name])
            d[name].grad = None
        return ret
```

I use deep copy because uninitialized_param._cleared or _zeroed flag cannot be shared, maybe.

**Comment by delta2323** [comment]
Hi, thank you for reporting the problem. I think this problem is same as #1615, although it is not worked on.

**Comment by niboshi** [comment]
Closing as this has been fixed in #2072.
